DBZ-3114 Support for Transaction Metadata in MySql connector

nitin456 opened the pull request.

gunnarmorling commented on debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/TransactionMetadataIT.java:
```diff
@@ -0,0 +1,88 @@
+                databaseName + ".customers", 2,
+                databaseName + ".orders", 1));
+    }
+}
```
Could you add a line-break at EOF.

nitin456 replied:
Addressed.

gunnarmorling reviewed (commented):
Nice, very happy to see it's that simple now as the connector has been moved over to the common connector framework. Just to things should be done:

* Add the feature to the connector docs, very similar to what's done for [Postgres](https://debezium.io/documentation/reference/connectors/postgresql#postgresql-transaction-metadata) (docs are under _documentation_ in the repo)
* Add the `io.debezium.config.CommonConnectorConfig.PROVIDE_TRANSACTION_METADATA` option to `MySqlConnectorConfig#CONFIG_DEFINITION`

nitin456 commented:
> * PROVIDE_TRANSACTION_METADATA

Will add the documentation and update the PR. Any specific reason to add PROVIDE_TRANSACTION_METADATA in MySqlConnectorConfig#CONFIG_DEFINITION? It's not done for Postgres.

gunnarmorling commented:
> It's not done for Postgres.

Yes, that's an oversight, it really should be there. I've logged https://issues.redhat.com/browse/DBZ-3132 for adding it for all the connectors which support this feature.

nitin456 commented:
> > It's not done for Postgres.
> 
> Yes, that's an oversight, it really should be there. I've logged https://issues.redhat.com/browse/DBZ-3132 for adding it for all the connectors which support this feature.

Fixed this and added documentation. Please review and merge if looks good.

gunnarmorling commented:
> I reverted the change of adding PROVIDE_TRANSACTION_METADATA in MySQLConnectorConfig. It was already added from CommonConnectorConfig. 

The field is _defined_ in `CommonConnectorConfig`, but specific sub-classes must explicitly add this to the fields they _expose_. See the Oracle connector config for how it's done: https://github.com/debezium/debezium-incubator/blob/master/debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java#L307 and https://github.com/debezium/debezium-incubator/blob/master/debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java#L397 (only that `configDef()` and `ALL_FIELDS` is shared for MySQL).

nitin456 commented:
> Test failures might be unrelated, there are some intermittent ones unfortunately atm.

I guess it was not related to intermittent failure. I was getting this error while running the unit test.
"Configuration provide.transaction.metadata is defined twice."

Do I need to exclude first and add this explicitly ?

gunnarmorling commented:
Ah, yes, that sounds related indeed. So looking at `MySqlConnectorConfig` again, it should be added - exactly once - to `CONFIG_DEFINITION`; is that what you tried?

gunnarmorling commented:
Raaah, no, now I realize this `CONFIG_DEFINITION` extends the parent one. Sorry, I had missed that :( So no, you don't need to add it, indeed it's inherited already. Sorry about the noise.

nitin456 commented:
> Ah, yes, that sounds related indeed. So looking at `MySqlConnectorConfig` again, it should be added - exactly once - to `CONFIG_DEFINITION`; is that what you tried?

Yes, like this 
https://github.com/nitin456/debezium/commit/a1541537da3629c57a29a02bff8255980fdce03a

One was already coming from HistorizedRelationalDatabaseConnectorConfig.CONFIG_DEFINITION.edit().

gunnarmorling commented:
> One was already coming from HistorizedRelationalDatabaseConnectorConfig.CONFIG_DEFINITION.edit().

Yes, exactly. I completely had missed we have this extension scheme in place already. All good then. Awaiting CI and then it should be ready to be merged. Sorry for the confusion.

nitin456 commented:
> > One was already coming from HistorizedRelationalDatabaseConnectorConfig.CONFIG_DEFINITION.edit().
> 
> Yes, exactly. I completely had missed we have this extension scheme in place already. All good then. Awaiting CI and then it should be ready to be merged. Sorry for the confusion.

No problem, thanks :)

gunnarmorling commented:
Applied; great work, thanks a lot!
